packages/edit-post/src/components/editor-initialization/index.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useContext } from '@wordpress/element';
+import { RegistryContext } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import {
+	blockSelectionListener,
+	adjustSidebarListener,
+	viewPostLinkUpdateListener,
+} from './listeners';
+
+/**
+ * Data component used for initializing the editor and re-initializes
+ * when postId changes or unmount.
+ *
+ * @param {number} postId
+ * @return {null} This is a data component so does not render any ui.
+ */
+export default function( { postId } ) {
+	const registry = useContext( RegistryContext );
+	useEffect( () => {
+		const blockSelectionUnsubscribe = registry.subscribe(
+			blockSelectionListener( registry )
+		);
+		const adjustSidebarUnsubscribe = registry.subscribe(
+			adjustSidebarListener( registry )
+		);
+		const viewPostLinkUpdateUnsubscribe = registry.subscribe(
+			viewPostLinkUpdateListener( registry )
+		);
+		registry.dispatch( 'core/nux' ).triggerGuide( [
+			'core/editor.inserter',
+			'core/editor.settings',
+			'core/editor.preview',
+			'core/editor.publish',
+		] );
+		return () => {
+			blockSelectionUnsubscribe();
+			adjustSidebarUnsubscribe();
+			viewPostLinkUpdateUnsubscribe();
+		};
+	}, [ registry, postId ] );
```

Comment:
I wonder if there's a potential flaw with this in that registry could be changing frequently with the internal state in the registry?  In which case it might be useful if registries had some sort of generated id that could be used as the dependency instead.  So essentially then we could do something like:

```js
const registry = useContext( RegistryContext );
useEffect( () => { /** existing stuff **/ }, [ registry.id, postId ] );
```